rita/src/rita_common/tunnel_manager/mod.rs

```diff
@@ -310,15 +318,17 @@ impl TunnelManager {
         let iface_index = if let Some(dev) = dev.clone() {
             KI.get_iface_index(&dev).unwrap()
         } else {
-            0
+            return Box::new(futures::future::err(TunnelManagerError::NoDevError.into()));
         };
 
+        let dev = dev.unwrap();
+
         Box::new(
             Connector::from_registry()
                 .send(actors::Resolve::host(their_hostname.clone()))
                 .from_err()
                 .and_then(move |res| {
-                    let url = format!("http://[{}%{:?}]:4876/hello", their_hostname, dev);
+                    let url = format!("http://[{}%{}]:4876/hello", their_hostname, dev);
```

Comment:
we don't actually use URL anywhere except for debug IIRC, so this shouldn't effect anything